core/utils/wei.go

```diff
@@ -53,19 +57,24 @@ func suffixExp(suf string) int32 {
 	}
 }
 
+// TODO: Consider moving this to `assets`
 // Wei extends Big to implement encoding.TextMarshaler and encoding.TextUnmarshaler with support for unit suffixes.
 type Wei Big
 
+func MaxWei(w, x *Wei) *Wei {
+	return NewWei(bigmath.Max(w.ToInt(), x.ToInt()))
+}
+
 // NewWei constructs a Wei from *big.Int.
 func NewWei(i *big.Int) *Wei {
 	return (*Wei)(i)
 }
 
-func (w *Wei) Cmp(w2 *Wei) int {
-	return (*big.Int)(w).Cmp((*big.Int)(w2))
+func NewWeiI[T constraints.Integer](i T) *Wei {
```

Comment:
Could `uint64` overflow?
```suggestion
func NewWeiI[T constraints.Signed](i T) *Wei {
```